### PR TITLE
Free resources in a try/finally

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs [hunk 1]
```diff
@@ -205,57 +205,62 @@ static void AddAll(HashSet<Checksum> checksums, ChecksumCollection checksumColle
             var usePool = missingChecksumsCount <= PooledChecksumArraySize;
             var missingChecksums = usePool ? s_checksumPool.Allocate() : new Checksum[missingChecksumsCount];
 
-            missingChecksumsCount = 0;
-            foreach (var checksum in checksums)
+            try
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs:208]: view with whitespace off.

**Review by ToddGrun — APPROVED** [review]
:shipit:

File: src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs (hunk 1 above)
Comment (sharwell) [reply src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs:208]: 💭 Seems like the pool could just be removed in favor of `stackalloc` for the initial array.

Comment (sharwell) [reply src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs:208]: Ah, nevermind due to callbacks

**Review by sharwell — APPROVED** [review]

File: src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs:208]: and async/await :(

**Pull request merged**